server/src/main/java/school/journal/service/impl/MarkService.java

```diff
@@ -85,136 +129,186 @@ public void delete(int id) throws ServiceException {
-        /*Criteria criteria = session.createCriteria(Mark.class);
-        criteria.createAlias("school_journal_db.pupil", "pupil", INNER_JOIN).
-                add(Restrictions.and(
-                        new MarkSpecificationBySubjectId(subjectId).toCriteria(),
-                        Restrictions.eq("pupil.class_id", classId)
-                ));*/
-        List<Mark> markList = null;
-        markList = session.createSQLQuery(
-                "SELECT  * " +
-                        "FROM  mark as m " +
-                        "JOIN pupil as p " +
-                        "ON m.pupil_id = p.pupil_id " +
-                        "WHERE p.class_id =" + classId + " " +
-                        "AND m.subject_id = " + subjectId + ";").list();
-        /*try {
+        try {
+            Criteria criteria = session.createCriteria(Mark.class);
+            criteria.createCriteria("pupil", INNER_JOIN).add(
+                    new PupilSpecificationByClassId(classId).toCriteria());
             markList = (List<Mark>) criteria.list();
             transaction.commit();
         } finally {
+            if (transaction.isActive()) {
+                transaction.rollback();
+            }
             session.close();
-        }*/
-        transaction.commit();
-        session.close();
+        }
         return markList;
     }
 
-    @Override
-    public List<Mark> getMarksForTermOrder(int classId) throws ServiceException {
+    private void validateClassIdBeforeSelect(int classId) throws ServiceException {
         try {
             validateId(classId, "Class");
         } catch (ValidationException exc) {
             LOGGER.error(exc);
             throw new ServiceException(exc);
         }
+    }
+
+    @Override
+    public List<Mark> getMarksForPupil(int pupilId) throws ServiceException {
+        validatePupilIdBeforeSelect(pupilId);
         Session session = sessionFactory.openSession();
         Transaction transaction = session.beginTransaction();
-        //TODO Learn how to create complex criteries
-        //TODO DON'T DELETE THE CODE BELOW
-        /*Criteria criteria = session.createCriteria(Mark.class);
-        criteria.createAlias("school_journal_db.pupil", "pupil", INNER_JOIN).
-                add(Restrictions.and(
-                        Restrictions.between("mark.date", startTerm, endTerm),
-                        Restrictions.eq("pupil.class_id", classId)
-                ));*/
         List<Mark> markList;
-        markList = session.createSQLQuery(
-                "SELECT  * " +
-                        "FROM  mark as m " +
-                        "JOIN pupil as p " +
-                        "ON m.pupil_id = p.pupil_id " +
-                        "WHERE p.class_id =" + classId + " ;").list();
-        /*try {
-            markList = (List<Mark>) criteria.list();
+        try {
+            Criteria criteria = session.createCriteria(Mark.class);
+            criteria.createCriteria("pupil", INNER_JOIN).add(
```

Comment:
i disagree with u. If we can use the same level of abstraction everywhere, we should do it.